tests/fft_test.py

```diff
@@ -138,6 +138,12 @@ def testFftn(self, inverse, real, shape, dtype, axes, s, norm):
       tol = 0.15
       jtu.check_grads(jnp_fn, args_maker(), order=2, atol=tol, rtol=tol)
 
+    # check dtypes
+    a = rng(shape, dtype)
+    dtype = jnp_fn(a).dtype
+    expected_dtype = jnp.result_type(0. if inverse and real else 0j, dtype)
+    self.assertEqual(dtype, expected_dtype)
```

Comment:
Could we cover this by setting `check_dtypes=True` in the Numpy check above?

Reply:
It seems that numpy always upcasts https://github.com/numpy/numpy/issues/17801

Reply:
I see... perhaps we could use something like the `_promote_like_jnp` decorator to address this? (I think we'd have to add a `complex=False` argument to it similar to the existing `inexact=False` to ensure complex output)

Reply:
hmm... this is certainly beyond my knowledge of jax :)

Reply:
Oh, this isn't really about JAX. I was just refering to a test utility that is used in some places, e.g. https://github.com/google/jax/blob/cf9a900d7879a1d8c7954c0687a5c764801d08df/tests/lax_numpy_test.py#L586

You could use this same test utility to modify `np.fft` so that its return type matches what is expected for JAX, and then we could set `check_dtypes=True` rather than having a separate block of code to check dtypes.

Reply:
If that's too much to bite off, then we can merge this as-is and I can clean up the test in a followup commit.

Reply:
That sounds good. Very interested in see how it's done in the followup commit to learn more about jax!